BUILDING.md

```diff
@@ -49,6 +49,31 @@ cmake --build build --config Release
 
 If you are using vcpkg as your C++ package manager, you might need to add `-DCMAKE_TOOLCHAIN_FILE=<path-to-your-vcpkd-root>/scripts/buildsystems/vcpkg.cmake` to the CMake command in the configuration step.
 
+### Unit Testing
+
+For configuring with testing enabled (only MSVC is enabled for now):
+
+```sh
+mkdir build
+cd build
+cmake .. --preset=test-msvc
+cmake --build test
+ctest -C debug test
```

Comment:
The other build instructions were crafter not to require `cd`ing into the `build` directory.  Is there a way to craft there instructions similarly for the configuration for testing?

Reply:
Can you help me understand the significance of avoiding the `cd`?

Reply:
> Can you help me understand the significance of avoiding the `cd`?

It is not about signficance of avoiding `cd`, but consistency of the instruction.

Reply:
Unfortunately `ctest` must be set to the working directory of the build tree: https://cmake.org/cmake/help/latest/manual/ctest.1.html#dashboard-client-via-ctest-command-line

I cannot see an option to manually set it.

Reply:
The `ctest` is after the configuration and build.  My comment affects the configuration step.

Here is what I expect the directory layout to be after build, and for testing:
    -- `build/`:  all the build output artefacts go here
    -- `build/test`: all the tests output artefacts go here

Does that match your view of what you would like to see?

If so, in the configuration step, I would have thought that something like `cmake -S test -B build/test --preset=test-msvc`   would work?

Reply:
Yes, I think that formulation will work and is consistent with the instructions above.